### PR TITLE
AEA-1280 Data request skill upgrades

### PR DESCRIPTION
**Pull request by jrriehl**
## Proposed changes

- Generalizes and renames `coin_price` skill to `advanced_data_request` (https://github.com/fetchai/agents-aea/issues/2230)
- Updates `fetch_beacon` skill to use `ledger_api` (https://github.com/fetchai/agents-aea/issues/2232)
- Updates `debug.md` with new location of `aea` binary

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/fetchai/skills/simple_oracle/strategy.py [hunk 1]
```diff
@@ -45,6 +46,9 @@ def __init__(self, **kwargs: Any) -> None:
         self._default_gas_deploy = kwargs.pop("default_gas_deploy", 0)
         self._default_gas_grant_role = kwargs.pop("default_gas_grant_role", 0)
         self._default_gas_update = kwargs.pop("default_gas_update", 0)
+        self._oracle_value_name = kwargs.pop(
+            "oracle_value_name", DEFAULT_ORACLE_VALUE_NAME
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/simple_oracle/strategy.py:50]: Input validation?

**Review by DavidMinarsch — COMMENTED** [review]
LGTM, some suggestions above

File: packages/fetchai/skills/fetch_beacon/handlers.py [hunk 2]
```diff
@@ -19,31 +19,29 @@
 
 """This package contains handlers for the fetch_beacon skill."""
 
-import json
-from typing import Any, Optional, cast
+from typing import Optional, cast, Dict, Any
 
 from vyper.utils import keccak256
```
[the remark this answers is not in the document]
Comment (jrriehl) [reply packages/fetchai/skills/fetch_beacon/handlers.py:24]: No, but it looks like the built-in `hashlib` package has it. I'll check to make sure it's the same and then replace.

[the remark this answers is not in the document]
Comment (jrriehl) [reply packages/fetchai/skills/fetch_beacon/handlers.py:24]: The `hashlib` implementation seems to be the wrong one, but it looks like there is a lighter-weight dependency that has this function: https://pypi.org/project/pysha3/. Worth switching to that one?

**Review by 5A11 — APPROVED** [review]
lgtm

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**